insights/specs/datasources/ssl_certificate.py

```diff
@@ -25,22 +34,37 @@ def httpd_certificate_info_in_nss(broker):
         SkipComponent: Raised when NSSEngine isn't enabled or "NSSCertificateDatabase" and
             "NSSNickname" directives aren't found
     """
-    conf = broker[HttpdConfTree]
+    confs = broker[LocalSpecs.httpd_files]
     path_pairs = []
-    virtual_hosts = conf.find('VirtualHost')
-    for host in virtual_hosts:
-        nss_engine = nss_database = cert_name = None
-        nss_engine = host.select('NSSEngine')
-        nss_database = host.select('NSSCertificateDatabase')
-        cert_name = host.select('NSSNickname')
-        if nss_engine and nss_engine.value and nss_database and cert_name:
-            path_pairs.append((nss_database[0].value, cert_name[0].value))
+    nss_engine = nss_database = cert_name = None
+    virtual_host_start = False
+    for conf in confs:
+        if os.path.exists(conf):
+            with open(conf) as a_f:
+                for line in a_f.readlines():
+                    line = line.strip()
+                    if line.startswith('#'):
+                        continue
+                    if line.startswith('<VirtualHost'):
+                        virtual_host_start = True
+                    if virtual_host_start:
+                        if line.startswith('NSSEngine'):
+                            nss_engine = line.split()[-1].lower().strip('"\'')
+                        if line.startswith('NSSCertificateDatabase'):
+                            nss_database = line.split()[-1].lower().strip('"\'')
+                        if line.startswith('NSSNickname'):
+                            cert_name = line.split()[-1].lower().strip('"\'')
+                        if line.startswith('</VirtualHost>'):
```

Comment:
@huali027 The code looks good to me in general. It could be more understandable if we can update the `if` and `elif` usage based on the logic, eg: 
```suggestion
                    if line.startswith('#'):
                        continue
                    elif line.startswith('<VirtualHost'):
                        virtual_host_start = True
                    if virtual_host_start:
                        if line.startswith('NSSEngine'):
                            nss_engine = line.split()[-1].lower().strip('"\'')
                        elif line.startswith('NSSCertificateDatabase'):
                            nss_database = line.split()[-1].lower().strip('"\'')
                        elif line.startswith('NSSNickname'):
                            cert_name = line.split()[-1].lower().strip('"\'')
                        elif line.startswith('</VirtualHost>'):
```
Same to `httpd_ssl_certificate_files` below.